### ISSUE TITLE
Documentation: `setup` and `teardown` aren't mentioned

### ISSUE THREAD

**Issue by alfredodeza** [pytest-dev/pytest#9549]
While browsing the [current release docs](https://docs.pytest.org/en/6.2.x/xunit_setup.html) on "xunit-style" support, specifically the usage of setup and tearing down (cleanup) of tests, I was surprised not to see `setup()` and `teardown()` mentioned. Instead, the documentation points out that for methods, one must use `setup_method()` and `teardown_method()`.

I've always used `setup()` and `teardown()` methods for setting up and cleaning up before and after each test method in a class for production code as well as teaching Pytest. 

Are those methods deprecated and shouldn't be used? Supported by mistake? Or is it an omission in the docs? Either way the docs need to be rectified for correctness. I'm happy to provide a PR with the right path.

**Comment by nicoddemus** [comment]
Hi @alfredodeza,

Fair question.

`setup` and `teardown` work because of the nose-style support.

For example, the example below works when executing `pytest`, but fails with `pytest -p no:nose`:

```python
class Test:

    def setup(self):
        self.x = 1

    def test_foo(self):
        assert self.x == 1
```

Perhaps we need to document `setup()` and `teardown()` more explicitly in the [nose page](https://docs.pytest.org/en/stable/nose.html)? 🤔

**Comment by alfredodeza** [comment]
Good point @nicoddemus . I forgot to mention I also went through the `nose` reference, both for Pytest as well on the Nose project. The Nose project doesn't mention `setup()` or `teardown()` methods at all [in their documentation](https://nose.readthedocs.io/en/latest/writing_tests.html#test-classes). But perhaps I'm missing some other section where those are explained?

Thanks for pointing out that these aren't pytest-native methods and are meant to support the other test framework (TIL!) 

In any case, I'm happy to provide a PR that explains this better, but would like to have something to correlate to Nose (couldn't find a reference)

**Comment by nicoddemus** [comment]
>  but would like to have something to correlate to Nose (couldn't find a reference)

Not sure either, I'm not familiar with nose's docs myself, but it certainly is supported:

```
λ pip list
Package    Version
---------- -------
nose       1.3.7
pip        21.3.1
setuptools 59.6.0
wheel      0.37.1

λ nosetests test_foo.py
.
----------------------------------------------------------------------
Ran 1 test in 0.000s

OK
```

**Comment by davidszotten** [comment]
i just came across this in a potentially even more surprising scenario. my test module had a (top level) import of a function that happened to be called `setup` 

(extra context: this is using django and pytest-django, that setup function happens to use the db. the test that usually calls it is marked with `django_db` but between pytest 6.2.5 (pytest-django 4.5.1) and 7.3.1, something's changed in the import order or otherwise causing the setup function to run before the marker is noted.

because the `setup` function is called by pytest, it is also the top of the stacktrace which adds to the confusion)

**Comment by davidszotten** [comment]
update: git bisect points to 89f0b5b5a2f485f7. not looked at detail but maybe it wasn't ok to "cheat a bit"?
